lib/file_store/base_store.rb

```diff
@@ -150,14 +150,23 @@ def cache_file(file, filename)
       dir = File.dirname(path)
       FileUtils.mkdir_p(dir) unless Dir.exist?(dir)
       FileUtils.cp(file.path, path)
-      # keep latest 500 files
+
+      # Keep latest 500 files
       processes = Open3.pipeline(
-        "ls -t #{CACHE_DIR}",
+        ["ls -t #{CACHE_DIR}", err: "/dev/null"],
         "tail -n +#{CACHE_MAXIMUM_SIZE + 1}",
         "awk '$0=\"#{CACHE_DIR}\"$0'",
         "xargs rm -f"
       )
-      raise "Error clearing old cache" if !processes.all?(&:success?)
+
+      ls = processes.shift
+
+      # Exit status `1` in `ls` occurs when e.g. "listing a directory
+      # in which entries are actively being removed or renamed".
+      # It's safe to ignore it here.
+      unless ls.exitstatus.in?([0, 1]) && processes.all?(&:success?)
```

Comment:
Might be easier to understand when using `if` instead of `unless`?

```suggestion
      if ![0, 1].include?(ls.exitstatus) || !processes.all?(&:success?)
```

Reply:
Do the two negations have a lower cognitive cost than `unless`? I don't know 😅 I'm ok with either one.

But why not `.in?`? 😛 I like how it reads naturally, similar to usual equality comparisons:

```rb
status == 1
status.in?([1, 2])
```

vs the reversed order:

```rb
1 == status
[1, 2].include?(status)
```

Reply:
`unless` feels always harder to understand when there’s more than 1 condition.

As for the not in? it’s just an habit I have to always put the “constants” on the left of a condition to prevent typoing = instead of ==.

Not really concerned about that though ;)